packages/oas-resolver/index.js

```diff
@@ -139,29 +140,53 @@ function testProtocol(input, backup) {
     return 'file:';
 }
 
-function resolveExternal(root, pointer, options, callback) {
-    var u = url.parse(options.source);
-    var base = options.source.split('\\').join('/').split('/');
+function getAbsoluteRefPath(source, $ref) {
+    let base = source.split('\\').join('/').split('/');
     let doc = base.pop(); // drop the actual filename
     if (!doc) base.pop(); // in case it ended with a /
+    base = base.join('/');
+    let u = url.parse(source);
+    let u2 = url.parse($ref);
+    let effectiveProtocol = testProtocol(u2.protocol, u.protocol)
+    if (effectiveProtocol === 'file:') {
+        return {
+            target: path.resolve(base ? base + '/' : '', $ref),
+            effectiveProtocol,
+            base
+        };
+    }
+    else {
+        return {
+            target: url.resolve(base ? base + '/' : '', $ref),
+            effectiveProtocol,
+            base
+        };
+    }
+}
+
+function resolveToName(ref, refs) {
+    let currentListOfResolved = [];
+    for (let key in refs) {
+        if (refs[key].resolvedAt) currentListOfResolved.push(refs[key].resolvedAt);
+    }
+
+    let potentialSchemaName = `#/components/schemas/${path.parse(path.basename(ref)).name}`;
+    if (currentListOfResolved.includes(potentialSchemaName)) {
+        let hash = crypto.createHash('sha1').update(ref).digest('hex');
+        return `#/components/schemas/${path.parse(path.basename(ref)).name}-${hash}`
```

Comment:
Did you decide against a configurable target location?

Reply:
Yes, we debated about where else it would go and the schema was really the only landing place if this feature is really about moving away from inline.